airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -521,6 +521,33 @@ def generate_config_file(
         :param eks_cluster_name: The name of the cluster to generate kubeconfig file for.
         :param pod_namespace: The namespace to run within kubernetes.
         """
+        args = ""
+        if self.region_name is not None:
+            args = args + f" --region-name {self.region_name}"
+
+        if self.aws_conn_id is not None:
+            args = args + f" --aws-conn-id {self.aws_conn_id}"
+
+        python_executable = f"python{sys.version_info[0]}.{sys.version_info[1]}"
```

Comment:
Can you leave a comment here explaining why this is necessary

Reply:
This is necessary because we need to determine which python executable the host is running in order to correctly call the `eks_get_token.py` script as we are doing below. By using `sys.version_info`, the solution is generic enough to work with any supported python version.